network/encoding.go

```diff
@@ -36,7 +36,11 @@ var ErrorType = MessageTypeID(uuid.Nil)
 
 // String returns the canonical string representation of the MessageTypeID
 func (mId MessageTypeID) String() string {
-	return uuid.UUID(mId).String()
+	t, ok := registry.get(mId)
+	if !ok {
+		return uuid.UUID(mId).String()
+	}
+	return t.String()
```

Comment:
If a type is already registered, you return the String representation of `reflect.Type` and if not a `uuid`? Is this for (human) readability or does it have another motivation?